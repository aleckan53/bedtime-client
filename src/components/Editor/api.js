import config from 'config'

export default {
  uploadStory(content) {
    return fetch(`${config.API_ENDPOINT}/stories`, {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${window.sessionStorage.getItem(config.TOKEN_NAME)}`,
        'content-type': 'application/json'
      },
      body: JSON.stringify({content})
    })
    .then(res => !res.ok ? res.json().then(err => Promise.reject(err)) : res.json()) 
  }
}