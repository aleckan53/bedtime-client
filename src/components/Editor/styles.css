.quill {
  min-height: 350px;
  height: 450px;
}

.ql-toolbar {
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  color: white;
  background: white;
}

.ql-container {
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  background: var(--btn-bg);
}

.editor {
  height: auto;
  position: relative;
  margin: 1rem 0 3rem;
  min-height: 400px;
}

.submit {
  position: absolute;
  bottom: -7rem;
  border: 1px solid var(--border-1);
  border-radius: 10px;
  height: 1.9rem;
  box-sizing: border-box;
  font-size: .9rem;
  font-weight: 600;
  margin-top: 1.2rem;
  box-shadow: 3px 5px 10px -5px var(--box-shadow);
  color: white;
  background: var(--main-color-1);
  left: 50%;
  transform: translateX(-50%);
  width: 50%;
}

@media (min-width: 750px) {
  .submit {
    width: 30%;
  }
}

.msg {
  position: absolute;
  top: 20%;
  height: 200px;
  text-align: center;
  border-radius: 10px;
  color: var(--main-text);
  display: grid;
  place-content: center;
  padding: 0 5%;
  box-sizing: border-box;
  left: 50%;
  transform: translateX(-50%);
}

.ql-editor.ql-blank::before {
  color: rgba(255,255,255,.7)
}

.ql-color .ql-color-label.ql-stroke {
  stroke: #00ff00;
}