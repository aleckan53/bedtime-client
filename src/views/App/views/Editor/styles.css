.quill {
  height: calc(80vh - 65px);
}

.ql-toolbar {
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
}

.ql-container {
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
}

.editor {
  height: 100%;
  position: relative;
  margin-bottom: 3rem;
}

.submit {
  position: absolute;
  bottom: 2rem;
}