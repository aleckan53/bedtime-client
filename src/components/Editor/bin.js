import React, { useState } from 'react'
import Api from './api'
import ReactQuill from 'react-quill'
import 'react-quill/dist/quill.snow.css'
import './styles.css'
import config from 'config'

const Editor = props => {

  const [state, setState] = useState({
    story: '',
    error: null,
  })
  const [loading, setLoading] = useState(false)
  const [message, setMessage] = useState('')

  const handleChange = (story) => {
    setState({
      ...state,
      story,
    })
  } 

  const handleSubmit = (e) => {
    e.preventDefault()
    setLoading(true)
    Api.uploadStory(state.story)
      .then(res => {
        setMessage(res.message)
        setState({
          story: '',
          error: null,
        })
        setLoading(false)
        setTimeout(() => {
          setMessage('')
        }, 4e3)
      })
      .catch(error => {
        setState({
          ...state,
          error
        })
      })
  }
  const userAuthorized = !!window.sessionStorage.getItem(config.TOKEN_NAME)

  const btnDisabled = userAuthorized && state.story.length > 800
  return (
    <>
      <section className='editor'>
        {!loading ? '' : <p className='msg'>Processing...</p>}
        <div className='msg'>{message}</div>
        <ReactQuill 
          theme='snow'
          value={state.story}
          onChange={handleChange}/>
      <button 
        className='submit' 
        onClick={handleSubmit} 
        disabled={!btnDisabled}>
        {userAuthorized ? 'Submit' : 'Please login'}
      </button>
      </section>
    </>
  )
}

export default Editor