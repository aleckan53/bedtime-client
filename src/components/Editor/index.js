import React, { useState } from 'react'
import ReactQuill from 'react-quill'
import config from 'config'
import Api from './api'
import 'react-quill/dist/quill.snow.css'
import './styles.css'

const Editor = props => {

  const [state, setState] = useState({
    html: '',
    error: null,
  })
  const [loading, setLoading] = useState(false)
  const [message, setMessage] = useState('')

  const handleChange = (html) => {
    setState({
      ...state,
      html,
    })
  } 

  const handleSubmit = (e) => {
    e.preventDefault()
    setLoading(true)
    Api.uploadStory(state.story)
      .then(res => {
        setMessage(res.message)
        setState({
          html: '',
          error: null,
        })
        setLoading(false)
        setTimeout(() => {
          setMessage('')
        }, 4e3)
      })
      .catch(error => {
        setState({
          ...state,
          error
        })
      })
  }
  const userAuthorized = !!window.sessionStorage.getItem(config.TOKEN_NAME)


  return (
    <section className='editor'>
      {!loading ? '' : <p className='msg'>Processing...</p>}
      <div className='msg'>{message}</div>
      <ReactQuill
        theme='snow'
        onChange={handleChange}
        value={state.html}
        bounds='.editor'
        modules={modules}
        formats={formats}
        placeholder={placeholder}
        > 
      </ReactQuill>
      <button className='submit' onClick={handleSubmit} disabled={!userAuthorized}>{userAuthorized ? 'Submit' : 'Please login'}</button>
    </section>
  )
}

export default Editor

const placeholder = `Please use H1 for story's title\n
Supported images types: *.png, *.jpeg. Others will be ignored\n
Story has to be at least of 800 characters long ~ 1 min read
`

const modules = {
  toolbar: [
    [{ 'header': '1'}, {'header': '2'}, { 'font': [] }],
    [{size: []}],
    ['bold', 'italic', 'underline'],
    [{'list': 'ordered'}, {'list': 'bullet'}, 
     {'indent': '-1'}, {'indent': '+1'}],
    ['image'],
    
  ],
  clipboard: {
    matchVisual: false,
  },
}

const formats = [
  'header', 'font', 'size',
  'bold', 'italic', 
  'list', 'bullet', 'indent',
  'image'
]