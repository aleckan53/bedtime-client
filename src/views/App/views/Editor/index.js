import React, { useState } from 'react'
import ReactQuill from 'react-quill'
import 'react-quill/dist/quill.snow.css'
import './styles.css'

const Editor = props => {

  const [state, setState] = useState('')

  const handleChange = (value) => {
    setState(value)
  } 

  const handleSubmit = () => {
    // Api submit
    // on succcess message
  }

  console.log(state)
  return (
    <section>
      <ReactQuill 
        theme='snow'
        value={state}
        onChange={handleChange}/>
        <button onClick={handleSubmit}>Preview</button>
    </section>
  )
}

export default Editor