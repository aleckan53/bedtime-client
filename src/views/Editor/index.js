import React, { useState } from 'react'
import ReactQuill from 'react-quill'
import 'react-quill/dist/quill.snow.css'
import './styles.css'

const Editor = props => {

  const [state, setState] = useState('')

  const handleChange = (value) => {
    setState(value)
  } 

  return (
    <section>
      <ReactQuill 
        theme='snow'
        value={state}
        onChange={handleChange}/>
    </section>
  )
}

export default Editor