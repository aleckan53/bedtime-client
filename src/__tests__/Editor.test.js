import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Editor from '../components/Editor'

describe(`Editor Component`, () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Editor />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})