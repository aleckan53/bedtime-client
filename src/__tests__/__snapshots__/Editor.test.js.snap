// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Component matches snapshot 1`] = `
<section
  className="editor"
>
  <div
    className="msg"
  />
  <Quill
    bounds=".editor"
    formats={
      Array [
        "header",
        "font",
        "size",
        "bold",
        "italic",
        "list",
        "bullet",
        "indent",
        "image",
      ]
    }
    modules={
      Object {
        "clipboard": Object {
          "matchVisual": false,
        },
        "toolbar": Array [
          Array [
            Object {
              "header": "1",
            },
            Object {
              "header": "2",
            },
            Object {
              "font": Array [],
            },
          ],
          Array [
            Object {
              "size": Array [],
            },
          ],
          Array [
            "bold",
            "italic",
            "underline",
          ],
          Array [
            Object {
              "list": "ordered",
            },
            Object {
              "list": "bullet",
            },
            Object {
              "indent": "-1",
            },
            Object {
              "indent": "+1",
            },
          ],
          Array [
            "image",
          ],
        ],
      }
    }
    onChange={[Function]}
    placeholder="Please use H1 for story's title

Supported images types: .png, .jpeg. Others will be ignored

Story has to be at least of 800 characters long ~ 1 min read
"
    theme="snow"
    value=""
  />
  <button
    className="submit"
    disabled={true}
    onClick={[Function]}
  >
    Please login
  </button>
</section>
`;
